### ISSUE TITLE
about mail address format as of 3.x

### ISSUE THREAD

**Issue by grundtein** [vert-x3/vertx-mail-client#174]
Hello,

I encountered a problem with the mailer when using 2-byte characters in the name of the email address.

### Version

3.x (Exact version is not known. i use it by quarkus mailer extension.)

### Context

example.
            "email": "to@mail.domain.com",
            "name": "亜亜亜亜亜亜亜亜亜亜亜亜"

I have prepared an email address in the form of String "亜亜亜亜亜亜亜亜亜亜亜亜 <to@mail.domain.com>" as input.
"vertx-mail-client" formats it in the following format.
to@mail.domain.com (亜亜亜亜亜亜亜亜亜亜亜亜)
In fact, mime line breaks will be inserted between '(' and ')'.

When I receive a mail in a mailer such as Outlook, i will see the following.
"亜亜亜亜�" "��亜亜亜亜亜亜�" "��" <to@mail.domain.com>

**Comment by gaol** [comment]
> I have prepared an email address in the form of String "亜亜亜亜亜亜亜亜亜亜亜亜 to@mail.domain.com" as input.

vertx-mail-client supports [3 ways of specifying email addresses](https://vertx.io/docs/vertx-mail-client/java/#_mailmessage_properties):

```
username@example.com
username@example.com (Firstname Lastname)
Firstname Lastname <username@example.com>
```

maybe you can try with input as: `亜亜亜亜亜亜亜亜亜亜亜亜 <to@mail.domain.com>` ?

**Comment by grundtein** [comment]
- username@example.com
- username@example.com (Firstname Lastname)
- Firstname Lastname <username@example.com>

I think the vertx-mail-client would format the input of all three methods above into the following format in io.vertx.ext.mail.mailencoder#encodeEmailList.
username@example.com (Firstname Lastname)

**Comment by gaol** [comment]
I finally understand this question from issue #215, it is the encoding problem on the 2-byte characters in the name, I assume the pr #217 will fix this issue as well.

**Comment by gaol** [comment]
I believe this has also been fixed via #217 , I am closing it. please re-open it if anyone face the same problem here.
